src/analysis/special_functions/trigonometric.lean

```diff
@@ -2802,7 +2802,7 @@ lemma cos_surjective : function.surjective cos :=
 begin
   intro x,
   obtain ⟨w, w₀, hw⟩ : ∃ w ≠ 0, 1 * w * w + (-2 * x) * w + 1 = 0,
-  { rcases exists_quadratic_eq_zero one_ne_zero (exists_eq_mul_self _) with ⟨w, hw⟩,
+  { rcases exists_quadratic_eq_zero (@one_ne_zero ℂ _ _) (exists_eq_mul_self _) with ⟨w, hw⟩,
```

Comment:
I have a hunch that there's a way to fix this (and break statements about associativity instead of statements about 1), which I'm trying out in https://github.com/leanprover-community/mathlib/tree/eric-wieser/semigroup_with_zero-fix.

Reply:
It worked, I've pushed a fix